src/Templating/EngineInterface.php

```diff
@@ -13,8 +13,55 @@
 
 namespace Sonata\BlockBundle\Templating;
 
-use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface as TemplatingEngineInterface;
+use Symfony\Component\HttpFoundation\Response;
 
-interface EngineInterface extends TemplatingEngineInterface
+/**
+ * @deprecated since 4.0, to be removed with 5.0.
+ */
+interface EngineInterface
 {
+    /**
+     * Renders a template.
+     *
+     * @param string $name       A template name or a TemplateReferenceInterface instance
+     * @param array  $parameters An array of parameters to pass to the template
+     *
+     * @throws \RuntimeException if the template cannot be rendered
+     *
+     * @return string The evaluated template as a string
+     */
+    public function render($name, array $parameters = []);
```

Comment:
Please add return type hints

Reply:
Added type hints and updated classes, implementing this interface